substrate/client/tracing/src/logging/event_format.rs

```diff
@@ -234,85 +211,22 @@ impl<'a> fmt::Display for FmtThreadName<'a> {
 //
 //       https://github.com/tokio-rs/tracing/blob/2f59b32/tracing-subscriber/src/fmt/time/mod.rs#L252
 mod time {
-	use ansi_term::Style;
 	use std::fmt;
 	use tracing_subscriber::fmt::{format, time::FormatTime};
 
-	pub(crate) fn write<T>(
-		timer: T,
-		writer: &mut format::Writer<'_>,
-		with_ansi: bool,
-	) -> fmt::Result
+	pub(crate) fn write<T>(timer: T, writer: &mut format::Writer<'_>) -> fmt::Result
 	where
 		T: FormatTime,
 	{
-		if with_ansi {
-			let style = Style::new().dimmed();
-			write!(writer, "{}", style.prefix())?;
+		if console::colors_enabled() {
+			write!(writer, "\x1B[2m")?;
 			timer.format_time(writer)?;
-			write!(writer, "{}", style.suffix())?;
+			write!(writer, "\x1B[0m")?;
 		} else {
 			timer.format_time(writer)?;
 		}
-		writer.write_char(' ')?;
-		Ok(())
-	}
-}
 
-/// A writer which (optionally) strips out terminal control codes from the logs.
-///
-/// This is used by [`EventFormat`] to sanitize the log messages.
-///
-/// It is required to call [`ControlCodeSanitizer::flush`] after all writes are done,
-/// because the content of these writes is buffered and will only be written to the
-/// `inner_writer` at that point.
-struct ControlCodeSanitizer<'a> {
-	sanitize: bool,
-	buffer: String,
-	inner_writer: format::Writer<'a>,
-}
-
-impl<'a> fmt::Write for ControlCodeSanitizer<'a> {
-	fn write_str(&mut self, buf: &str) -> fmt::Result {
-		self.buffer.push_str(buf);
-		Ok(())
-	}
-}
-
-// NOTE: When making any changes here make sure to also change this function in `sp-panic-handler`.
-fn strip_control_codes(input: &str) -> std::borrow::Cow<str> {
```

Comment:
We should not remove this code.

Reply:
Or is `console` taking care of this?

Reply:
According to https://github.com/console-rs/console/blob/master/src/utils.rs#L625-L632
When `console` renders a styled string, it will check the global `colors_enabled()` and `colors_enabled_stderr()`.
If false, it will skip rendering ANSI control codes.

So I think `console` taking care of this.

There is only one restriction we need to ensure we only use `console` for the ANSI styling.

Reply:
About the global settings, check https://github.com/console-rs/console/blob/master/src/utils.rs#L30-L75

Reply:
is this change OK?

Reply:
Then the test that you removed should still work?

Reply:
I'm not sure what you mean.
Do you mean the case that runtime panicked?

Reply:
<img width="1571" alt="image" src="https://github.com/user-attachments/assets/06cbcae4-9d18-4af8-af43-d516efe770fc">

Reply:
https://github.com/paritytech/polkadot-sdk/pull/2923/files#r1685265036 I mean this one.

Reply:
I have a question:
Why need to call `strip_control_codes(msg)` in `panic_hook()` of `sp-panic-handler`?
The `PanicInfo#payload()` should be the argument of `panic!`. When will it contain ANSI control codes?

Reply:
AFAIR it is just a protection against someone being able to trigger a panic and putting ANSI control codes into it. "It is possible", but very unlikely.